src/shared/components/query/QueryStore.ts

```diff
@@ -297,8 +230,28 @@ export class QueryStore
 	{
 		let ids:string[] = this._selectedStudyIds.keys();
 		const selectableStudies = this.selectableStudiesSet;
-		ids = ids.filter(id=>!!selectableStudies[id]);
-		return this.forDownloadTab ? ids.slice(-1) : ids;
+		let selectableIds = ids.filter(id=>!!selectableStudies[id]);
+
+		if(ids.length !== selectableIds.length) {
+			let entities = this.caseIds.trim().split(/\s+/g);
+			let normStudies:string [] = entities
+			.map(entity=>{

```

Comment:
this should be a reduce function which can then map and filter at the same time and won't produce the undefined items.